[14.0][MIG] asterisk_click2dial

alexis-via opened the pull request.

Commit:
Initial import.
Long life to the asterisk_click2dial module for OpenERP !

Commit:
- smaller "Dial" button on partner contact
- option to allow the use of phone numbers in National format,
  as discussed with Albert. This option is on the definition of the server.
  -> required a large change of the reformat number process
- added exact link to documentation in module description
- some clean-up

Commit:
Vianney changed the name of the 'alert-info' field to 'alert_info'...
but he also made the change in the name of the SIP header !
This commit reverts this mistake.

Commit:
[FIX] WaitTime parameter is now really taken into account
In fact, it corresponds to the 'Timeout' instruction for the Asterisk Manager

Commit:
Make error messages displayed on screen translatable.
Add French translation for these messages.

Commit:
- Add copyright headers in XML files
- Convert the comments for the functions to the standard style

Commit:
Country_prefix is now a "required" field.
Added "active" field on asterisk.server object.
In order to access all the Asterisk servers, included the desactivated ones,
I added a view dedicated to the object asterisk.server, which is accessible
under Administration > Configuration.

Commit:
Check that the user has a callerID.
Thanks to Bernat Brunet from NaN for pointing this out.

Commit:
First implementation of get_name_from_phone_number function
Make phone and mobile fields searchable
Took into account some pylint coding convention suggestions

Commit:
Much better function get_name_from_phone_number :
- we now handle the fact that phone numbers may contain spaces or special chars such in OpenERP, which is often the case in real life
- check input number

Commit:
Add AGI script and its wrapper to manage timeout.

Commit:
Don't put the admin user by default, it's not a good example.
Added a comment about the user rights.

Commit:
Add a group "Asterisk CallerID" decidated to the CallerID name lookup feature
Update module description.

Commit:
Better context handling
Update in comments

Commit:
Update URL of the module documentation.

Commit:
[FIX] There shouldn't be any space between the name and the phone number in the callerID instruction to Asterisk.
Patch courtesy of Marcel van der Boom. Thanks for your feedback !

Commit:
- Send CallerID to Asterisk as UTF-8 by default
  Accoring to my tests and the info I got, SIP phones support UTF-8 fine
- Create a command line option to convert the name from UTF-8 to ASCII (-a)
- add default parameter to the help message

Commit:
Networking code is now IPv6-ready

Commit:
[FIX] Avoid a view conflict with the "base_contact" module by adding a priority on the res.partner view.
Better handling of check functions when there are several ids.

Commit:
[FIX] Base_contact also inherit the view base.view_partner_address_form1 and removes the "mobile" field, so I need to change the priority to this view too.

Commit:
[FIX] Wrong if condition fixed
Thanks to Todd Johnson for pointing out this mistake.

Commit:
Avoid code duplication in views : the full tree/form view for Asterisk server now inherit from the tree/form view in the company form.

Commit:
Changed author name to Akretion
Changed view names
Great solution (thanks to Sebastien for finding it !) to avoid having 2 different tree&form views for asterisk server.

Commit:
Add support for XML-RPC secure with SSL.

Commit:
Fallback to utf-8 if sys.stdout.encoding fails.

Commit:
Create a dedicated Tab in the res_users view.
Larger phone field in res_partner_address (smaller 'dial' button)

Commit:
Start to clean the networking code :
. adds debug message
. starts to better manage character encoding in the networking code... but I still need more info about AMI on this.

Commit:
Large code cleanup :
- now parses answers received from Asterisk and display errors to the user
- now properly encodes the requests sent to Asterisk in ASCII (AMI doesn't support other charsets unfortunately)
- required fields are now required=True
- Single constraint function for all asterisk.server fields
- Fields that shouldn't have non-ascii caracters are not validated if they have non-ASCII caracters (including CallerID)
- clean-up error message code
- if no Asterisk server is defined for a user, it takes the first Asterisk serve
r defined
- methods now always return something

Commit:
Cosmetic changes.

Commit:
Convert to new logger API of OpenERP 6.1
Remove print

Commit:
Add translations from extra-trunk

Commit:
[FIX] Fix issue in constraint function for res_users.

Commit:
[IMP] Better AGI script :
- cleaner code
- should avoid these Asterisk errors :
   utils.c ast_carefulwrite: write() returned error: Broken pipe

Commit:
New feature : add wizard that opens the partner form of the calling party

Commit:
Remove purchase menu entry, because it would require a dependancy on the 'purchase' module.

Commit:
Update the module description to add the new feature "open calling partner".
Code clean-up.

Commit:
Improve (and fixes) the 'Open calling partner' function : better usability and less clicks
Update the description.

Commit:
Add 2 screenshots.

Commit:
For the feature "Open calling party", if the phone number is not in OpenERP, he proposes to create a new partner with this phone or mobile number.

Commit:
Add user-specific alert-info option.
Ability to have multiple alert-info options (separate them with '|').

Commit:
Add logo that will appear in the list of modules in web client

Commit:
Convert the phone number to international format before writing it on the new partner.
Factorize code.

Commit:
Works even if national_prefix is empty.

Commit:
Update translation file.
Update French translation.
Update module description.

Commit:
Add 'variable' field on res.users
New design of the pop-up "open calling partner" when the calling number is not found
When calling number is not found, now also proposes to update an existing contact
Update French translation
Update module description

Commit:
[FIX] avoid a crash when emptying "contact to update"
Fix typo

Commit:
Better error message.

Commit:
Port asterisk_click2dial_crm to version 6.1 :
- code re-factoring
- add "open CRM call" from the "open calling partner" wizard

Commit:
asterisk_click2dial_crm :
- when you do a click2dial, it now triggers a wizard that proposes to create a CRM phone call (instead of directly creating the CRM phone call)
- Add option on res_users for users who don't want to trigger the wizard when they do a click2dial
- Add FR translation
- update module description and module category

asterisk_click2dial :
- update module category

Commit:
Patch courtesy of daemon dog :
- add support for SCCP phones
- adds dial_suffix to allow some tweaking for SCCP phones
- move menu entry of asterisk servers to Settings >Configuration > Asterisk servers
- strip (0) from phone number in all scenarios

Added by me : re-org of res.users form view, to separate standard parameters from advanced/optional ones.

Commit:
Fix crahes when using the Web interface (these crashes did not occur with the Gtk Client)

Commit:
Don't set readonly fields in the "open calling partner" wizard, to avoid calling default_get twice (and thus connect to asterisk twice)
Make "open calling partner" work with Asterisk 1.4 (or it is IAX trunk ?)

Commit:
Add Local/ as possible channel type

Commit:
Now uses the phonenumber lib to reformat numbers in "clean" format and store them in E.164 format. Start to modify the code to take advantage of this great lib.
Thanks to Ludovic Gasc for pointing this lib to me during PyconFR 2012 in Paris !

Fix a bug with Open Calling Partner when CallerID has non-ASCII chars.

Move all code of wizard to wizard directory.

IMPORTANT : when you upgrade to this revision, run the wizard "Reformat all phone numbers" from Settings > Configuration > Asterisk.

Commit:
Add support for geolocalisation via the phonenumbers lib in the get_cid_name.py script. If you add --geoloc, the numbers that are not found in OpenERP will be geolocalized.

Thanks to Ludovic Gasc for making me discover this great "phonenumbers" lib.

Commit:
Display the country name only when it's not my own country. This make the option to display only the city useless IMHO, so I removed this option.

Commit:
Add debug log, to help a user.
Small code clean-up.

Commit:
More debug and warning messages, to help a user in its debug.

Commit:
[FIX] open calling partner should work again.

Commit:
Now use a PATH variable in get_cid_name_timeout.sh
get_cid_name.py : don't update the CallerID name if there is already one present.

Commit:
Use py-asterisk from http://code.google.com/p/py-asterisk/ (up-to-date mercurial version required) instead of our own code
Add a field 'resource' on res.users, which is used in "SIP/resource", instead of "SIP/internal_number", which was not correct. YOU NEED TO UPDATE YOUR DATA !

Commit:
Add some exclusion on CallerID for which we don't query openerp or try to geolocate.

Commit:
Remove the import of the socket lib, that we don't use any more.

Commit:
Port to OpenERP 7.0
Add logo for the module asterisk_click2dial_crm

Commit:
Oups, fix my previous commit.

Commit:
In the wizards, Instead of creating a new record and opening it, I open a new form with the values. It's better when using the Web interface, because it avoids a click on "Edit".

Commit:
Add a country to "Your company" in demo data (required for this module).

Commit:
[FIX] res.partner.address -> res.partner

Commit:
Update views to take advantage of the new styles provided by OpenERP 7.0.

Commit:
Remove <field name="type">form/tree</field> in `ir.ui.view` model (deprecated in OpenERP 7.0)

Commit:
Cosmetic changes.

Commit:
The number of digits to match from the end of the phone number is now configurable. This code is now fully located in the asterisk_click2dial module (removed from the "get_cid_name.py" script).
WARNING : if you upgrade your asterisk_click2dial module to this version, you should also update the script "get_cid_name.py" on your Asterisk server to this version.
The imports now use the new openerp paths.
Convert self.pool.get('obj') to self.pool['obj'] so that it crashes exactly where it should.

Commit:
Country prefix now has an intelligent default value.
Just for fun, because I love the phonenumbers lib !

Commit:
Update version number of the script get_cid_name.py, to make it easier to identify people who forgot to update it.
Update year in copyright

Commit:
Remove Local/ from the list of chan types, because Originate doesn't work with Local/phonenumber.

Commit:
Add encoding in the shell script get_cid_name_timeout.sh
Remove an unused line of code.

Commit:
Add read rights on asterisk.server for group "Asterisk CallerID"
Patch courtesy of Marcel van der Boom (HS-Development BV)
Convert ACLs from XML format to CSV format.

Commit:
Avoid a crash when the answer to the "Status" query doesn't contain what is expected.
Patch courtesy of Marcel van der Boom (HS-Development BV).

Commit:
[FIX] Adapt view inheritance of res.partner form to post-v7.0-release changes
Patch courtesy of Marcel van der Boom (HS-Development BV)

Commit:
Add compatibility with Asterisk 1.4 for the function "Open calling partner"

Commit:
[IMP] Add Dial button in Lead and Opportunity views
[IMP] Add CDR account management for Originating Call

Commit:
Move the generic code from res.partner to a new object asterisk.common.
res.partner and crm.lead now inherit from asterisk.common. Thanks to Sebastien Beau for giving me the idea.

Commit:
[IMP] move generic_reformat_phonenumbers function to asterisk.common

Commit:
[IMP] Add Dial button in Claim Form View

Commit:
Add back "Local" as channel type.

Commit:
Fix a few strings.
Update POT file and French translation.

Commit:
The regular "phone", "mobile" and "fax" fields now store the phone number in E164 format ; the special fields phone_e164, mobile_e164 and fax_164 are not used any more.
Add phone widget that handle the "nice" display of phone number to the user.
IMPORTANT : you need to restart the wizard "Reformat all phone numbers" in order to put the "phone", "mobile" and "fax" fields in E164 format (without spaces).

Commit:
No need for ilike to match numbers !

Commit:
Add 'external_dependencies' in __openerp__.py
Remove unneeded imports

Commit:
FIX to make the script work with the latest version of the phonenumbers lib

Commit:
Use RFC 2806 : add a fax: URL and convert callto: URLs to tel: URLs
The wizard "Reformat phones numbers" is now extended to objects other than res.partner
Update coding standard
Don't use xpath in inherited views when not necessary

Commit:
Update my script to make it work with the latest versions of the phonenumbers lib.

Commit:
Long-awaited feature : automatic pop-up on incoming calls in the OpenERP Web interface. Special thank to Jean-Sébastien Suzanne for his collaboration on this new feature.

Commit:
Extract Asterisk-independant code (the widget="phone" + phone number reformatting) from the asterisk_click2dial module to a new base_phone module.

Commit:
Add option try_parent in open_filtered_object
In the wizard "Open Calling Partner", when you click on "open related orders" for example, it will open the orders with an explicit filter that you can then delete.

Commit:
Very large changes in the structure of the project :
- move more asterisk-independant code to base_phone module and create new asterisk-independant modules
- rename module asterisk_popup to base_phone_popup
- add ability to get callerid names not only from partners, but also from leads, employees, applicants, etc...
- same for auto pop-up : not only for partners, but also for leads, employees, etc...

Commit:
Add support for update of the name on OUTGOING calls (for far, it was only possible on incoming calls).

Commit:
Fix documentation.

Commit:
Add a "Test Connection to Asterisk" button on the asterisk.server form view.

Commit:
The "Dial" button is now inside the phone widget : this is great because it avoids the double inherit of views with phone fields (4 view XML files are removed in asterisk_click2dial* modules thanks to this !)
The "Open Calling Partner" feature is now located in the top right menu, next to the Preferences (instead of the Sales menu)
A click on tel: and fax: links doesn't open a new tab any more
Update description and copyrights of asterisk_click2dial_crm

Commit:
Update POT file and partial update of FR translation
Convert regular string to translatable string

Commit:
Port to OpenERP v8.0

Commit:
Better button for Open Calling Partner

Commit:
Add href="#", so that the mouse cursor has the right aspect when over the phone icon
Simplify HTML

Commit:
Feature "Open Calling Partner" replaced by "Open Caller", with a completely new behavior
As a consequence, the modules asterisk_click2dial_event and asterisk_click2dial_crm_claim have been dropped
New wizard "Number not found" (inspired by the old "Open Calling Partner" with many enhancements), which is called by base_phone_popup and Open Caller
Formatting of number for click2dial is now handled by the phonenumbers lib: as a consequence, 3 fields have been dropped from asterisk.server (national_prefix, international_prefix, country_prefix)
New option 'dial_button_invisible' in the phone widget
New sample script asterisk_click2dial/scripts/openerp_popup_timeout.sh
Update translations and some module descriptions
All the modules are now PEP8 compliant

Commit:
FIX remplace debug stuff by real code !

Commit:
Use =like instead of like in phone number search, because I add the '%' myself
Add logger.error in the function used by the inherit of write/create, for easier debugging when it is called by a webservice
Add . in PATH in my sample shell scripts
Code cleanup (thank you flake8 !)

Commit:
Don't fail if Asterisk and asterisk_click2dial aren't installed

Commit:
Fix code style

Commit:
Ability to avoid the 'Not found in Odoo' as callerID name when phone is not found

Commit:
[FIX] Bug #11 AMI test: success is now only reported when it really works !

Commit:
PEP8 fix

Commit:
Update screenshots
Remove old-style icons on buttons

Commit:
Highlight dial button in screenshot

gurneyalex committed:
Add OCA as author of OCA addons

In order to get visibility on https://www.odoo.com/apps the OCA board has
decided to add the OCA as author of all the addons maintained as part of the
association.

yvaucher committed:
Move module icon in static/description

Commit:
Add support for "Create CRM phonecall" on leads (and not only partners)
Re-organise the codebase to prepare the arrival of a new module for a telephony system other than Asterisk

Commit:
Remove an import that is not needed any more

vaab committed:
new: pkg: adds requirements.txt to explicit dependencies and ease scripted installations.

Commit:
FIX old/new API call

Commit:
Please welcome the new module ovh_telephony_connector !
Some code clean-up and re-organisation in the generic modules
Prepare the removal of the module asterisk_click2dial_crm

alexis-via committed:
set_name_agi.py: add support for JSON-RPC and JSON-RPC SSL

PEP8

sbidoul committed:
[UPD] prefix versions with 8.0

pedrobaeza committed:
[MIG] Make modules uninstallable

alexis-via committed:
Port asterisk_click2dial to new API

Commit:
Port JS and controller code of asterisk_click2dial to v9

alexis-via committed:
Port crm_phone to v9

Code cleanup, fixes, small enhancements
The code to extract the calling party number from the list of channels is now inheritable

alexis-via committed:
Restore the object crm.phonecall that was dropped in v9 community

Proper name_get for callerid for applicants
Small bugfixes/enhancements

alexis-via committed:
Add notes about a workaround to make the phone logo in top bar work with Odoo enterprise

alexis-via committed:
Fix bug #93... but it seems do_notify() is not working in v9 in this branch

pedrobaeza committed:
[MIG] Make modules uninstallable

pedrobaeza committed:
[MIG] Rename manifest files

alexis-via committed:
Port crm_phone, hr_phone, hr_recruitment_phone, event_phone to v10

Remove crm_claim_phone, because there is no crm_claim module in v10
Start to port asterisk_click2dial (JS code still needs to be ported)

Commit:
[FIX] base_phone: Fixed click2dial feature

alexis-via committed:
Port opencaller to v10

alexis-via committed:
PEP8 fix

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

Commit:
[10.0][ADD] asterisk_click2dial: PJSIP channel type

alexis-via committed:
Allow to work with asterisk server not company-related

oca-transbot committed:
OCA Transbot updated translations from Transifex

alexis-via committed:
Improve AGI script using a dedicated lib

oca-transbot committed:
OCA Transbot updated translations from Transifex

sbidoul committed:
remove obsolete .pot files [ci skip]

oca-travis committed:
[UPD] Update asterisk_click2dial.pot

alexis-via committed:
Port to v12 base_phone, asterisk_click2dial, crm_phone, hr_phone, event_phone and hr_recruitment_phone

alexis-via committed:
Improve script: better default port, works when incoming phone number has a +

oca-travis committed:
[UPD] Update asterisk_click2dial.pot

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

alexis-via committed:
Forward port dial button refactor from v11 to v12

Move tests from base_phone to crm_phone (to benefit from crm_phone_validation)
Don't use a demo partner in test (this partner was removed in later versions)
Continue to remove dependancy on demo data in tests
Forward port of base_phone_popup from v10 to v12 (use web_notify)

oca-travis committed:
[UPD] Update asterisk_click2dial.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: connector-telephony-12.0/connector-telephony-12.0-asterisk_click2dial
Translate-URL: https://translation.odoo-community.org/projects/connector-telephony-12-0/connector-telephony-12-0-asterisk_click2dial/

max3903 committed:
[FIX] asterisk_click2dial: Travis errors

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 20.9% (19 of 91 strings)

Translation: connector-telephony-12.0/connector-telephony-12.0-asterisk_click2dial
Translate-URL: https://translation.odoo-community.org/projects/connector-telephony-12-0/connector-telephony-12-0-asterisk_click2dial/pt_BR/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 78.0% (71 of 91 strings)

Translation: connector-telephony-12.0/connector-telephony-12.0-asterisk_click2dial
Translate-URL: https://translation.odoo-community.org/projects/connector-telephony-12-0/connector-telephony-12-0-asterisk_click2dial/pt_BR/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (91 of 91 strings)

Translation: connector-telephony-12.0/connector-telephony-12.0-asterisk_click2dial
Translate-URL: https://translation.odoo-community.org/projects/connector-telephony-12-0/connector-telephony-12-0-asterisk_click2dial/pt_BR/

Theyiot committed:
CO-3343 Open caller button works properly again

oca-travis committed:
[UPD] Update asterisk_click2dial.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: connector-telephony-12.0/connector-telephony-12.0-asterisk_click2dial
Translate-URL: https://translation.odoo-community.org/projects/connector-telephony-12-0/connector-telephony-12-0-asterisk_click2dial/

mimusica committed:
Added translation using Weblate (Flemish)

mimusica committed:
Translated using Weblate (Flemish)

Currently translated at 45.1% (41 of 91 strings)

Translation: connector-telephony-12.0/connector-telephony-12.0-asterisk_click2dial
Translate-URL: https://translation.odoo-community.org/projects/connector-telephony-12-0/connector-telephony-12-0-asterisk_click2dial/nl_BE/

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: connector-telephony-12.0/connector-telephony-12.0-asterisk_click2dial
Translate-URL: https://translation.odoo-community.org/projects/connector-telephony-12-0/connector-telephony-12-0-asterisk_click2dial/

oca-travis committed:
[UPD] Update asterisk_click2dial.pot

OCA-git-bot committed:
asterisk_click2dial 12.0.1.0.1

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: connector-telephony-12.0/connector-telephony-12.0-asterisk_click2dial
Translate-URL: https://translation.odoo-community.org/projects/connector-telephony-12-0/connector-telephony-12-0-asterisk_click2dial/

alexis-via committed:
[MIG] asterisk_click2dial from v12 to v14

PJSIP is now the default channel type instead of SIP
Convert README to the new subdir format
Minor code improvements

alexis-via committed:
asterisk_click2dial: black, isort, ...

cvinh commented:
Hello @alexis-via 
I tested your last commit.
It appears that the window from where we click to dial closes by itself and go back to the Discuss page
Do you have this issue too ?
Best regards

alexis-via commented:
Yes, we need to fix that. Don't hesitate to propose a PR to this branch to fix it.

cvinh committed:
[FIX] We have to stay on the same page after clicking on the link

Gretel5X commented on asterisk_click2dial/models/asterisk_server.py:
```diff
@@ -0,0 +1,283 @@
+        if not user.asterisk_chan_type:
+            raise UserError(_("No channel type configured for the current user."))
+
+        # We check if the current user has an internal number
```
The comment does not really correspond to the error message, I'd either say `resource` or `internal number`.
Maybe it is clear for pros on Odoo/Asterisk?

alexis-via replied:
Agree, I removed the comment, which doesn't add real value.

Gretel5X commented on asterisk_click2dial/models/asterisk_server.py:
```diff
@@ -0,0 +1,283 @@
+
+    @api.model
+    def _get_calling_number_from_channel(self, chan, user):
+        """Method designed to be inherited to work with
```
Inconsistent use of `"""`, above the text was after a newline and the closing `"""`as well

alexis-via replied:
fixed

Gretel5X commented on asterisk_click2dial/models/asterisk_server.py:
```diff
@@ -0,0 +1,283 @@
+    @api.model
+    def get_record_from_my_channel(self):
+        calling_number = self.env["asterisk.server"]._get_calling_number()
+        # calling_number = "0641981246"
```
Is that a valid number for testing? Maybe remove it :)

Gretel5X replied:
I just realised that that number is there since several years, so nevermind my comment

alexis-via replied:
ok. It just help me locate the place to put my test number.

Commit:
Merge pull request #7 from invitu/14.0-fix_base_phone

[FIX] We have to stay on the same page after clicking on the link

alexis-via commented:
I have ported the AGI script to py3, I'll push it tonight.

alexis-via committed:
asterisk_click2dial: migrate AGI script to python3

Full code cleanup of the AGI script
Minor modifications to take into account the remark of the reviewers

alexis-via commented:
Travis is green! If everyone is ok, I'll merge tomorrow.

alexis-via commented:
/ocabot merge nobump
